extensions/color-picker/src/name-color.ts

```diff
@@ -0,0 +1,66 @@
+import { Clipboard, closeMainWindow, launchCommand, LaunchType, showHUD } from "@raycast/api";
+import { callbackLaunchCommand } from "raycast-cross-extension";
+import { addToHistory } from "./history";
+import { Color, PickColorCommandLaunchProps } from "./types";
+import { getFormattedColor } from "./utils";
+import namer from "color-namer";
+import { pickColor } from "swift:../swift/color-picker";
+import { showFailureToast } from "@raycast/utils";
+
+export default async function command(props: PickColorCommandLaunchProps) {
+  await closeMainWindow();
+
+  try {
+    const pickedColor = (await pickColor()) as Color | undefined;
+    if (!pickedColor) {
+      return;
+    }
+
+    addToHistory(pickedColor);
+
+    const hex = getFormattedColor(pickedColor);
+    if (!hex) {
+      throw new Error("Failed to format color");
+    }
+    const colorNames = namer(hex);
+    if (colorNames.basic.length < 1) {
+      throw new Error("Failed to get color name");
+    }
+    const colorName = colorNames.basic[0].name;
+
+    if (props.launchContext?.callbackLaunchOptions) {
+      if (props.launchContext?.copyToClipboard) {
+        await Clipboard.copy(colorName);
+      }
+
+      try {
+        await callbackLaunchCommand(props.launchContext.callbackLaunchOptions, { hex });
+      } catch (e) {
+        await showFailureToast(e);
+      }
```

Comment:
We can callback with both `hex` and `colorName` here. The API documentation needs to be updated as well.

```suggestion
      try {
        await callbackLaunchCommand(props.launchContext.callbackLaunchOptions, { hex, colorName });
      } catch (e) {
        await showFailureToast(e);
      }
```

We can also add the `colorName` callback payload to `pick-color` command.

I can help if you need any help on composing documentation and `callLaunchCommand` functions.

Reply:
We should make the API convenient. So let's return both `hex` and `colorName`.

Reply:
fine, I mod as you advice